NuRadioMC/simulation/simulation.py

```diff
@@ -147,112 +1095,134 @@ def __init__(self, inputfilename,
-                logger.error("ice model is set to 'custom' in config file but no custom ice model is provided.")
-                raise AttributeError("ice model is set to 'custom' in config file but no custom ice model is provided.")
-            self._ice = ice_model
-        else:
-            self._ice = medium.get_ice_model(self._cfg['propagation']['ice_model'])
-
-        self._mout = collections.OrderedDict()
-        self._mout_groups = collections.OrderedDict()
-        self._mout_attrs = collections.OrderedDict()
+        # initialize detector simulation modules
+        # due to mostly historical reasons, the detector simulation is provided as protected member functions,
+        # the user inherits from the simulation class and provides the detector simulation functions as protected
+        # member functions. This has the advantage that the user has potential access to all internal member variables
+        # of the simulation class and thereby full control. However, this is discouraged as it makes it difficult to assure
+        # backwards compatibility.
+        # The default behaviour is that the user provides a function that defines the signal chain (i.e. filter and amplifiers etc.)
+        # and another function that defines the trigger. 
+        # The user can also provide a function that defines the full detector simulation (i.e. the signal chain and the trigger) including 
+        # how noise is added etc.
+        # The user needs to either provide the functions `_detector_simulation_filter_amp` and `_detector_simulation_trigger` or
+        # the function `_detector_simulation_part1` and `_detector_simulation_part2`.
+        self.detector_simulation_part1 = None
+        self.detector_simulation_part2 = None
+        self.detector_simulation_filter_amp = None
+        self.detector_simulation_trigger = None
+        if hasattr(self, '_detector_simulation_filter_amp'):
+            self.detector_simulation_filter_amp = self._detector_simulation_filter_amp
+        if hasattr(self, '_detector_simulation_trigger'):
+            self.detector_simulation_trigger = self._detector_simulation_trigger
+        if hasattr(self, '_detector_simulation_part1'):
+            self.detector_simulation_part1 = self._detector_simulation_part1
+        if hasattr(self, '_detector_simulation_part2'):
+            self.detector_simulation_part2 = self._detector_simulation_part2
+        if((self.detector_simulation_filter_amp is None) ^ (self.detector_simulation_trigger is None)):
+            logger.error("Please provide both detector_simulation_filter_amp and detector_simulation_trigger or detector_simulation_part1 and detector_simulation_part2")
+            raise ValueError("Please provide both detector_simulation_filter_amp and detector_simulation_trigger or detector_simulation_part1 and detector_simulation_part2")
+        if((self.detector_simulation_part1 is None) ^ (self.detector_simulation_part2 is None)):
```

Comment:
first time this syntax of using `(..)` in if conditions is used.

Reply:
I don't understand this comment.

Reply:
Before, the syntax `if statement:` is used. This and in the following sometimes this syntax is used `if(statement):` is used.